OrbitService/SshWatchdog.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_SERVICE_SSH_WATCH_DOG_H_
+#define ORBIT_SERVICE_SSH_WATCH_DOG_H_
+
+#include <atomic>
+#include <chrono>
+#include <optional>
+#include <string>
+
+// This class reads stdin and signals a shutdown via `exit_requested` in the
+// following situations.
+// 1. When EOF arrives on stdin
+// 2. When the it has been activated and no more data arrived on stdin for a
+// duration of at least `timeout_in_seconds` seconds.
+//
+// The Watchdog is activated upon reading `start_passphrase` on stdin.
+class SshWatchdog {
+ public:
+  explicit SshWatchdog(std::atomic<bool>* exit_requested);
+
+  void LoopTick();
+
+ private:
+  std::optional<std::string> ReadStdIn();
```

Comment:
What is the semantic difference between an empty string and an empty optional here?

Reply:
true, just returning string (potentially empty) now instead of optional